sofascoreAcademy/sofascoreAcademy/ViewModel/EventsViewModel.swift

```diff
@@ -7,14 +7,34 @@
 
 import Foundation
 import UIKit
-import SofaAcademic
 import SnapKit
 
-class EventsViewModel {
+class EventsViewModel: ObservableObject {
+    enum State {
+        case idle
+        case loaded([LeagueSection])
+        case loading
+        case error
+    }
+    @Published private(set) var state: State = .idle
+    
     var sections: [LeagueSection] = []
     func loadSections() async {
-        let events = Homework3DataSource().events()
-        sections = getLeagueSections(for: events)
+       
+        self.state = .loading
+        do {
+            let events = try await APIClient.getEvents(sport: MenuViewModel.shared.selectedSport.urlSlug)
+            sections = getLeagueSections(for: events)
+            
+            if sections.isEmpty {
+                self.state = .error
+            } else {
```

Comment:
Ovo zapravo nije error, ako imaš neki date picket/ kalendar, moguće da za neki dan nemaš sportova i onda ako je prazna lista prikažeš npr. message "No events on this day".
Error je više ako ti pukne request ili tako nešto.
Možeš ostaviti kako je sada.